asserts/snapasserts/validation_sets.go

```diff
@@ -281,3 +353,91 @@ func (v *ValidationSets) Conflict() error {
 	}
 	return nil
 }
+
+// CheckInstalledSnaps checks installed snaps against the validation sets.
+func (v *ValidationSets) CheckInstalledSnaps(snaps []*InstalledSnap) error {
+	installed := naming.NewSnapSet(nil)
+	for _, sn := range snaps {
+		installed.Add(sn)
+	}
+
+	// snapName -> validationSet key -> validation set
+	invalid := make(map[string]map[string]bool)
```

Comment:
we don't seem to be storing false anywhere, so maybe all those `map[string]bool` could be `map[string]struct{}` for lower memory usage?

Reply:
Indeed, OTOH we seem to prefer map[string]bool everywhere (under similar circumstances), with only a few cases of map[..]struct{} in the entire code. Given the fact that this is a temporary map and validation sets are going to be used for a handful of snaps, I'm not sure it would win us anything. Of course I'm happy to change it in a followup if we find it's worth it.